### ISSUE TITLE
False positive when returning an array of tuples

### ISSUE THREAD

**Issue by Dschee** [realm/SwiftLint#595]
Hi, I've just come across a documented method which returns an array of tuples where the valid_docs rule is showing a warning. I've double-checked everything and there seems to be no issue with the documentation part – Xcode is showing the correct documentation when I Alt-Click the method.

Here's a demo method where I could reproduce the issue:

``` Swift
/// Create an array of tuples for demoing an issue.
///
/// - Returns: An array of integer tuples.
public func createTuples() -> [(Int, Int)] {
    return []
}
```

This shows the following warning:
`warning: Valid Docs Violation: Documented declarations should be valid. (valid_docs)`

This is, I think, a false positive and should be fixed.

**Comment by ioveracker** [comment]
This seems to be related to using a tuple as a generic type parameter (e.g. `Array<(Int, Int)>`). I'm running into this with a return type of `Observable<(String, NSURL)>`. Using a typealias for the tuple is a workaround currently.

**Comment by marcelofabri** [comment]
Closing this since we removed this rule.
